WidgetLibrary/WidgetLibrary/Controls/Counter.ascx.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+namespace WidgetLibrary.Controls
+{
+    public partial class Counter : System.Web.UI.UserControl
+    {
+        public int Count
+        {
+            get
+            {
+                object o = ViewState["Count"];
+                if (o != null)
+                {
+                    return (int)o;
+                }
+                else
+                {
+                    return 0;
+                }
+            }
+            set
+            {
+                ViewState["Count"] = value;
+            }
+        }
+
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            if (!IsPostBack)
+            {
+                CounterLabel.Text = Count.ToString();
```

Comment:
I'd change this `Page_Load` event handler to a `Page_PreRender` event handler:

```csharp
protected void Page_PreRender(object sender, EventArgs e)
{
    CounterLabel.Text = Count.ToString();
}
```

Setting the `CounterLabel.Text` on every page render would allow you to remove that code from the button click event handlers.